src/dosext/mfs/util.c

```diff
@@ -167,6 +167,14 @@ char *strupperDOS(char *src)
   return s;
 }
 
+char *strnupperDOS(char *src, int n)
+{
+  char *s = src;
+  for (; *src && n; src++, n--)
```

Comment:
```suggestion
  for (; n > 0 && *src; src++, n--)
```

to avoid problems with negative `n` and to avoid dereferencing beyond the requested number of characters.

Reply:
okay cool, thanks